### ISSUE TITLE
Codec tests are bogus

### ISSUE THREAD

**Issue by zcorpan** [WebPlatformTest/HTML5test#19]
It seems the codec tests give the wrong results in Opera. Opera does not support MP3 or AAC.

```
            canPlayType: function(t) {
                var r = this.element.canPlayType(t)
                return r == 'maybe' || r == 'probably';
            },
```

FYI, Opera will return 'maybe' for MIME types it doesn't know about (since gstreamer might be able to play it -- to know for sure you'd need to try to play it and listen for the error event for failure or loadedmetadata event for success).

```
            t2: function() {
                var r = !!this.element.canPlayType && this.canPlayType('video/mp4; codecs="avc1.42E01E, mp4a.40.2"');
                this.section.setItem('H.264 codec support', r)
```

This is testing for MPEG-4/H.264 baseline/AAC low-complexity. If you just want to test H.264 baseline in MPEG-4 container, use 'video/mp4; codecs="avc1.42E01E"'

```
                var r = !!this.element.canPlayType && this.canPlayType('video/ogg; codecs="theora, vorbis"');
```

Same here. (Nit: Ogg is a container format, so saying "Ogg Theora codec" is not entirely accurate.)

```
                var r = !!this.element.canPlayType && this.canPlayType('audio/mpeg3;');
```

audio/mpeg3 is not a registered MIME type. audio/mpeg is the correct MIME type for MP3.

```
            t4: function() {
                var r = !!this.element.canPlayType && this.canPlayType('audio/x-m4a;');
```

audio/x-m4a is not a registered MIME type. audio/mp4; codecs="mp4a.40.2" is the correct MIME type for AAC low-complexity in MPEG-4 container.

```
            t5: function() {
                var r = !!this.element.canPlayType && this.canPlayType('audio/wav; codecs="1"');
```

(audio/wav is not registered, but there's no registered MIME type for WAVE so browsers implemented audio/wav, so this check is fine.)

```
                this.section.setItem('WAV codec support', r)
```

(Nit: WAVE is container format, PCM is codec.)

**Comment by zcorpan** [comment]
I see at least audio/mpeg was fixed already in http://github.com/NielsLeenheer/html5test/blob/master/index.html but some of the other comments still apply

**Comment by NielsLeenheer** [comment]
Hi Simon,

Thanks for your comments. I've updated the codec tests and fixed the problems you identified. 

First of all a codec is now only reported as supported when the canPlayType returns 'probably'. Secondly I only test for video and not longer the audio codec when detecting video codec support. And I changed the non standard AAC MIME types to audio/mp4 with the correct codec.

But I also made another change for detecting MP3. As you probably are well aware of, the audio/mpeg MIME type is not just for MP3, but also other codecs that are part of the MPEG standard. That is why browsers that support MP3 only return 'maybe' when asked if they support audio/mpeg.

I now use 'audio/mpeg; codecs="mp3"' for detecting MP3 support. I haven't found any documentation that this is how it should work, but it seems to work fine in Safari (reporting 'probably' instead of 'maybe'), but Chrome still reports 'maybe'. Is using this codec okay? Or should I just drop it accept 'maybe' as MP3 support in any case.

In any case I am also looking into detecting MP3 support using a small audio sample and listening for the loadedmetadata event. In that case the previous point becomes moot.

Cheers,

Niels
